### ISSUE TITLE
Grafana is failed to start with MySQL 8.0 as a database

### ISSUE THREAD

**Issue by ballab1** [grafana/grafana#12581]
### What Grafana version are you using?
       4.* and 5.*

### What datasource are you using?
from my *grafana.ini*

       #################################### Database ####################################
       [database]
       type = mysql
       host = 10.1.3.10:3306
       name = grafana
       user = anon
       password = "password"
       max_idle_conn = 2
       max_open_conn = 20
### What OS are you running grafana on?
       Alpine 3.6 thru Alpine 3.8
### What did you do?
       started up container with following volumes mapped
```
            restart: unless-stopped
            volumes:
                - ./grafana/etc:/etc/grafana
                - ./grafana/log:/var/log/grafana
                - ./grafana/var_lib:/var/lib/grafana 
```
### What was the expected result?
       container starts and stays up
### What happened instead?
       container continuously restarts because of the following error:

```
t=2018-07-11T15:04:28-0400 lvl=dbug msg="Skipping migration: Already executed" logger=migrator id="create index UQE_org_user_org_id_user_id - v1"
t=2018-07-11T15:04:28-0400 lvl=dbug msg="Skipping migration: Already executed" logger=migrator id="Update org table charset"
t=2018-07-11T15:04:28-0400 lvl=dbug msg="Skipping migration: Already executed" logger=migrator id="Update org_user table charset"
t=2018-07-11T15:04:28-0400 lvl=dbug msg=Executing logger=migrator sql="UPDATE org_user SET role = 'Viewer' WHERE role = 'Read Only Editor'"
t=2018-07-11T15:04:28-0400 lvl=info msg="Executing migration" logger=migrator id="Migrate all Read Only Viewers to Viewers"
t=2018-07-11T15:04:28-0400 lvl=eror 
                           msg="Executing migration failed" 
                           logger=migrator 
                           id="Migrate all Read Only Viewers to Viewers" 
                           error="Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'role = 'Viewer' WHERE role = 'Read Only Editor'' at line 1"
t=2018-07-11T15:04:28-0400 lvl=eror 
                           msg="Exec failed" 
                           logger=migrator 
                           error="Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'role = 'Viewer' WHERE role = 'Read Only Editor'' at line 1" sql="UPDATE org_user SET role = 'Viewer' WHERE role = 'Read Only Editor'"
t=2018-07-11T15:04:29-0400 lvl=eror 
                           msg="Server shutdown" 
                           logger=server 
                           reason="Service init failed: Migration failed err: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'role = 'Viewer' WHERE role = 'Read Only Editor'' at line 1"
```
container then restarts, continues loading and restarts again at same place

After searching code, the SQL 
```
      UPDATE org_user SET role = 'Viewer' WHERE role = 'Read Only Editor';
````
is located in '/pkg/services/sqlstore/migrations/org_mig.go'

Tried this from mysql prompt and got the same error. 
```
      mysql> UPDATE org_user SET role = 'Viewer' WHERE role = 'Read Only Editor';
      ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'role = 'Viewer' WHERE role = 'Read Only Editor'' at line 1
```
read manual and realized that the problem is due to conflict between column name 'role' and reserved keyword 'role'
changed command to avoid conflict 
```
      mysql> UPDATE org_user SET org_user.role = 'Viewer' WHERE org_user.role = 'Read Only Editor';
      Query OK, 0 rows affected (0.00 sec)
      Rows matched: 0  Changed: 0  Warnings: 0
```

Version of MySQL: 
```
      Server version: 8.0.0-dmr MySQL Community Server (GPL)
      Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.
      Oracle is a registered trademark of Oracle Corporation and/or its
      affiliates. Other names may be trademarks of their respective
      owners.
 ```

Also:
Been waiting months to receive my confirmation email from Grafana Community.

**Comment by torkelo** [comment]
Very strange , this has worked with all MySQL servers we tested with (and on our CI sever), will check version 8

**Comment by alexanderzobnin** [comment]
This is due to changes in MySQL 8.0 (https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-0.html):
> MySQL now supports roles, which are named collections of privileges.

> Note
> ROLE now is a reserved word and cannot be used as an identifier without identifier quoting.

**Comment by Turing-Chu** [comment]
I received the same error
```
t=2018-07-22T02:58:22+0800 lvl=info msg="App mode production" logger=settings
t=2018-07-22T02:58:22+0800 lvl=info msg="Initializing SqlStore" logger=server
t=2018-07-22T02:58:22+0800 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=mysql
t=2018-07-22T02:58:22+0800 lvl=info msg="Starting DB migration" logger=migrator
t=2018-07-22T02:58:22+0800 lvl=eror msg="Server shutdown" logger=server reason="Service init failed: Migration failed err: this authentication plugin is not supported
```

```
CentOS Linux release 7.4.1708

grafana  Version 5.2.1 (commit: 2040f61)

mysql  Ver 8.0.11 for linux-glibc2.12 on x86_64 (MySQL Community Server - GPL)

just start grafana after import grafana data from mysql-5.6 to mysql-8.0 
```

**Comment by thatsparesh** [comment]
@alexanderzobnin, @ballab1  It looks like mysql did acknowledge this as a bug on their end, and revoked ROLE from being a reserved keyword
Reference 
- https://bugs.mysql.com/bug.php?id=83274 - Fixed in 8.0.1.
- https://dev.mysql.com/doc/refman/8.0/en/keywords.html - ROLE; became nonreserved in 8.0.1

This was an issue for a period of time (in 8.0.0), but it soon became non-reproducible.

**Comment by alexanderzobnin** [comment]
@thatsparesh thanks for reply, it seems, we could close this without any changes in Grafana. 
/cc @torkelo
